struct.js

```diff
@@ -249,7 +249,7 @@ ThriftStruct.prototype.createConstructor = function createConstructor(name, fiel
         source += '    this.' + field.name + ' = null;\n';
         source += '    if (options && typeof options.' + field.name + ' !== "undefined") ' +
             '{ this.' + field.name + ' = options.' + field.name + '; }\n';
-        if (field.defaultValue !== null) {
+        if (field.defaultValue !== null && field.defaultValue !== undefined) {
```

Comment:
This can be captured by != null, which is effectively !== null && !== undefined.